django/contrib/admin/options.py

```diff
@@ -1389,6 +1389,19 @@ def get_changeform_initial_data(self, request):
                 initial[k] = initial[k].split(",")
         return initial
 
+    def _get_obj_does_not_exist_redirect(self, request, opts, object_id):
+        """
+        Create a message informing the user that the object doesn't exist
+        and return a redirect to the admin index page.
+        """
+        msg = _('%(name)s object with primary key %(key)s does not exist.') % {
```

Comment:
IMO, `%r` might be an improvement (to avoid quoting integers) except for the `u''` prefix that would be added for strings on Python 2...